lib/puma/server.rb

```diff
@@ -424,7 +425,7 @@ def process_client(client, buffer)
 
             ThreadPool.clean_thread_locals if clean_thread_locals
 
-            unless client.reset(@status == :run)
+            unless client.reset(@status == :run && @fast_track_ka_timeout)
```

Comment:
I'm loathe to cache anything unless we actually have a benchmark proving it is a significant speed difference. In this case, making a comparison against an integer.

I'm not even sure max_threads is always static in all cases.